### PR TITLE
Utilise Eddsa & Public Keys - Part 4

### PR DESCRIPTION
**Pull request by biscuitdey**
# Description

Use Eddsa signatures in circuitInputParser and computeCircuitInputs.

## Related Issue

#759 

## Motivation and Context

Use multiple types of signatures 

## How Has This Been Tested

Running all existing tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Request to be added as a Code Owner/Maintainer

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] I commit to abide by the Responsibilities of Code Owners/Maintainers.

### CODE REVIEW

**Review by Therecanbeonlyone1969 — APPROVED** [review]
LGTM

**Review by ognjenkurtic — APPROVED** [review]

File: examples/bri-3/src/bri/auth/guards/didJwt.guard.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ export class DidJwtAuthGuard implements CanActivate {
     context: ExecutionContext,
   ) {
     // TODO: store did in bpi subject and remove constant
-    const didSubstrLength = 13;
+    const didSubstrLength = 20;
```
Comment (Therecanbeonlyone1969) [inline examples/bri-3/src/bri/auth/guards/didJwt.guard.ts:44]: @ognjenkurtic @biscuitdey is this recorded as an issue? Cannot see it.

File: examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/utils/computePublicInputs.spec.ts [hunk 2]
```diff
@@ -9,10 +9,11 @@ describe('Compute public inputs utils', () => {
     //Generate signature
     const message = 'Test';
     const messageHash = Buffer.from(
-      ethers.utils.arrayify(ethers.utils.hashMessage(message)),
+      ethers.toBeArray(ethers.hashMessage(message)),
     );
     const signature = await wallet.signMessage(message);
-    const expandedSignature = ethers.utils.splitSignature(signature);
+    //const expandedSignature = ethers.utils.splitSignature(signature);
```
Comment (Therecanbeonlyone1969) [inline examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/utils/computePublicInputs.spec.ts:15]: @ognjenkurtic @biscuitdey still needed as a placeholder? If not remove.

File: examples/bri-3/test/sriUseCase.e2e-spec.ts [hunk 3]
```diff
@@ -169,7 +168,7 @@ describe('SRI use-case end-to-end test', () => {
   });
 
   it('Waits for a single VSM cycle and then verifies that the transaction has been executed and that the state has been properly stored', async () => {
-    await new Promise((r) => setTimeout(r, 20000));
+    await new Promise((r) => setTimeout(r, 50000));
```
Comment (Therecanbeonlyone1969) [inline examples/bri-3/test/sriUseCase.e2e-spec.ts:171]: @ognjenkurtic @biscuitdey better in constant file?

**Review by Therecanbeonlyone1969 — APPROVED** [review]
LGTM. Left two minor comments. No holdup for approval and merge

**Pull request merged**